File: test/lit/zig/comparison.zig
// RUN: %cot emit-cir %s | %FileCheck %s

// Feature #007: Comparison — Zig syntax

pub fn isEqual(a: i32, b: i32) i32 {
    if (a == b) return 1;
    return 0;
}

pub fn isLess(a: i32, b: i32) i32 {
    if (a < b) return 1;
    return 0;
}

// CHECK-LABEL: func.func @isEqual
// CHECK: cir.cmp eq
// CHECK: cir.condbr

// CHECK-LABEL: func.func @isLess
// CHECK: cir.cmp slt
// CHECK: cir.condbr

File: test/lit/swift/comparison.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: comparison operators

func isLess(a: Int32, b: Int32) -> Bool {
    return a < b
}

func isEqual(a: Int32, b: Int32) -> Bool {
    return a == b
}

func isGreater(a: Int32, b: Int32) -> Bool {
    return a > b
}

// CHECK-LABEL: func.func @isLess
// CHECK: cir.cmp slt, %{{.*}}, %{{.*}} : i32

// CHECK-LABEL: func.func @isEqual
// CHECK: cir.cmp eq, %{{.*}}, %{{.*}} : i32

// CHECK-LABEL: func.func @isGreater
// CHECK: cir.cmp sgt, %{{.*}}, %{{.*}} : i32
